### PR TITLE
fix: Flaky test

### PR DESCRIPTION
**Pull request by tcc-sejohnson**
What it says on the tin 😬 

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it. (sort of, I guess?)

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

File: packages/kit/test/apps/basics/test/test.js [hunk 1]
```diff
@@ -864,7 +864,7 @@ test.describe('Actions', () => {
 			const log_promise = page.waitForEvent('console');
 			const button = page.locator(`#${id}`);
 			await button.click();
-			expect(await button.textContent()).toBe('processed'); // needed to make sure action completes
+			await page.waitForSelector(`button#${id}[data-processed="true"]`);
```
Comment (gtm-nayan) [inline packages/kit/test/apps/basics/test/test.js:867]: Can you reuse the locator here?

I don't remember the exact method, but there should be something like toHaveAttribute
```suggestion
			await expect(button).toHaveAttribute('data-processed', 'true')
```

Comment (gtm-nayan) [reply packages/kit/test/apps/basics/test/test.js]: It was `toHaveAttribute` indeed.

**Comment by gtm-nayan** [comment]
ughh lint failure, every damn time I commit from github

File: packages/kit/test/apps/basics/test/test.js [hunk 2]
```diff
@@ -864,7 +864,7 @@ test.describe('Actions', () => {
 			const log_promise = page.waitForEvent('console');
 			const button = page.locator(`#${id}`);
 			await button.click();
-			expect(await button.textContent()).toBe('processed'); // needed to make sure action completes
+			await expect(button).toHaveAttribute('data-processed', 'true')
```
Comment (gtm-nayan) [inline packages/kit/test/apps/basics/test/test.js:867]:
```suggestion
			await expect(button).toHaveAttribute('data-processed', 'true');
```

Comment (tcc-sejohnson) [reply packages/kit/test/apps/basics/test/test.js]: Will this wait for it to change from `false` to `true`?

Comment (gtm-nayan) [reply packages/kit/test/apps/basics/test/test.js]: It'll throw if it doesn't become true within a timeout, 5 secs IIRC.

**Review by gtm-nayan — APPROVED** [review]

File: packages/kit/test/apps/basics/test/test.js [hunk 3]
```diff
@@ -828,50 +828,53 @@ test.describe('Matchers', () => {
...
-			old_name: 'data',
-			new_name: 'formData',
-			call_location: 'use:enhance submit function'
-		},
-		{
-			id: 'access-data-in-callback',
-			old_name: 'data',
-			new_name: 'formData',
-			call_location: 'callback returned from use:enhance submit function'
+	test.only(`Accessing v2 deprecated properties results in a warning log`, async ({
```
Comment (gtm-nayan) [inline packages/kit/test/apps/basics/test/test.js:831]:
```suggestion
	test(`Accessing v2 deprecated properties results in a warning log`, async ({
```

**Pull request merged**